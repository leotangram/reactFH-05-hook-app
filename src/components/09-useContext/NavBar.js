import React from 'react'
import { NavLink } from 'react-router-dom'

const NavBar = () => {
  return (
    <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
      <NavLink exact activeClassName="active" className="navbar-brand" to="/">
        useContext
      </NavLink>
      <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
        <div className="navbar-nav">
          <NavLink
            exact
            activeClassName="active"
            className="nav-item nav-link"
            to="/"
          >
            Home <span className="sr-only">(current)</span>
          </NavLink>
          <NavLink
            exact
            activeClassName="active"
            className="nav-item nav-link"
            to="/about"
          >
            About
          </NavLink>
          <NavLink
            exact
            activeClassName="active"
            className="nav-item nav-link"
            to="/login"
          >
            Login
          </NavLink>
        </div>
      </div>
    </nav>
  )
}

export default NavBar
