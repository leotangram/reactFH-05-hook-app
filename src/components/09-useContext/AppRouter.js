import React from 'react'
import { BrowserRouter, Switch, Route, Redirect } from 'react-router-dom'
import NavBar from './NavBar'
import AboutScreen from './AboutScreen'
import LoginScreen from './LoginScreen'
import HomeScreen from './HomeScreen'

const AppRouter = () => {
  return (
    <BrowserRouter>
      <div>
        <NavBar />
        <div className="container">
          <Switch>
            <Route exact path="/" component={HomeScreen} />
            <Route exact path="/about" component={AboutScreen} />
            <Route exact path="/login" component={LoginScreen} />
            <Redirect to="/" />
          </Switch>
        </div>
      </div>
    </BrowserRouter>
  )
}

export default AppRouter
