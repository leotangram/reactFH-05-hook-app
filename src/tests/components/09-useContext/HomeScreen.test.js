import React from 'react'
import { mount } from 'enzyme'
import HomeScreen from '../../../components/09-useContext/HomeScreen'
import { UserContext } from '../../../components/09-useContext/UserContext'

describe('<HomeScreen />', () => {
  const user = {
    name: 'Leonardo',
    email: 'leonardo@gmail.com'
  }

  const wrapper = mount(
    <UserContext.Provider value={{ user }}>
      <HomeScreen />
    </UserContext.Provider>
  )
  test('should show correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
