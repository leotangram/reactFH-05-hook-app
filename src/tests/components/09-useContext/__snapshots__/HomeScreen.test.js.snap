// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> should show correctly 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "Leonardo",
   "email": "leonardo@gmail.com"
}
  </pre>
</div>
`;
